app/src/main/java/com/google/droidjump/MainActivity.java

```diff
@@ -25,48 +25,54 @@
 import android.content.SharedPreferences;
 import android.os.Bundle;
 import androidx.appcompat.app.AppCompatActivity;
+import com.google.droidjump.leveldata.LevelConfig;
+import com.google.droidjump.leveldata.LevelConfigParser;
+import com.google.droidjump.leveldata.LevelStrategy;
+import java.util.ArrayList;
 
 /**
  * Represents main activity.
  */
 public class MainActivity extends AppCompatActivity {
-    private int levelsCount;
+    private int levelsLastIndex;
     private SharedPreferences gameData;
     private SharedPreferences.Editor gameDataEditor;
+    private ArrayList<LevelConfig> gameLevels;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         gameData = getSharedPreferences(GAME_VIEW_DATA, Context.MODE_PRIVATE);
         gameDataEditor = gameData.edit();
-        levelsCount = GameConstants.GAME_LEVELS_COUNT;
+        gameLevels = LevelConfigParser.getLevelConfigsFromResource(R.raw.level_configs, this);
+        levelsLastIndex = gameLevels.size() - 1;
         setContentView(R.layout.main_activity);
     }
 
-    public int getLevelsCount() {
-        return levelsCount;
+    public int getLevelsLastIndex() {
+        return levelsLastIndex;
     }
 
     public void onCurrentLevelCompleted() {
-        int currentLevel = getCurrentLevel();
-        int lastLevel = getLastLevel();
-        if (currentLevel < levelsCount) {
+        int currentLevelIndex = getCurrentLevelIndex();
+        int lastLevel = getLastLevelIndex();
+        if (currentLevelIndex < levelsLastIndex) {
             // Increasing the current level
-            gameDataEditor.putInt(GAME_VIEW_CURRENT_LEVEL_STRING, ++currentLevel);
+            gameDataEditor.putInt(GAME_VIEW_CURRENT_LEVEL_STRING, ++currentLevelIndex);
             // Increasing the last level
-            if (currentLevel > lastLevel) {
-                gameDataEditor.putInt(GAME_VIEW_LAST_LEVEL_STRING, currentLevel);
+            if (currentLevelIndex > lastLevel) {
+                gameDataEditor.putInt(GAME_VIEW_LAST_LEVEL_STRING, currentLevelIndex);
             }
             gameDataEditor.apply();
         }
     }
 
-    public void setCurrentLevel(int currentLevel) {
+    public void setCurrentLevelIndex(int currentLevel) {
         gameDataEditor.putInt(GAME_VIEW_CURRENT_LEVEL_STRING, currentLevel);
         gameDataEditor.apply();
     }
 
-    public void setLastLevel(int lastLevel) {
+    public void setLastLevelIndex(int lastLevel) {
```

Comment:
can this method be private?

Reply:
I refactored this method from Maksym's code and it is never used, so I deleted this method